cpp/whatsapp-with-vonage/src/main.cc

```diff
@@ -0,0 +1,160 @@
+        for (const std::string& varName : envVarNames) {
+            context.log(varName.c_str());
+            const char* varValue = std::getenv(varName.c_str());
+            if (varValue == nullptr) {
+                context.error(varName.c_str());
+                missingVars.push_back(varName);
+            }
+        }
+
+        if (!missingVars.empty()) {
+            context.error("Error: Missing environment variables:");
+            for (const std::string& missingVar : missingVars) {
+                std::cerr << " " << missingVar;
+            }
+            std::cerr << std::endl;
+            throw std::runtime_error("Missing environment variables.");
+        }
+    }
+
+    static RuntimeOutput main(RuntimeContext& context) {
+        RuntimeRequest req = context.req;
+        RuntimeResponse res = context.res;
+
+        context.log("Hello, Logs!");
+        // checkEnvVars(context, {
+        //                           "VONAGE_API_KEY",
+        //                           "VONAGE_API_SECRET",
+        //                           "VONAGE_API_SIGNATURE_SECRET",
+        //                           "VONAGE_WHATSAPP_NUMBER",
+        //                       });
+        // // If something goes wrong, log an error
+        context.error("Hello, Errors!");
+        // context.log(serveStaticFile(context, "index.html"));
+        std::vector<std::string> files = listFiles("/usr/local");
+        // std::vector<std::string> files = listFiles("/usr/local");
+        for (const std::string& file : files) context.log(file);
+        if (req.method == "GET") {
+            std::string indexHtml = getStaticFile("index.html");
+            context.log(indexHtml);
+            return res.send(indexHtml);
+        }
+
+        Json::Value payload = stringToJson(req.bodyRaw);
+        context.log(payload.toStyledString());
+
+        std::vector<std::string> requiredFields = {"from", "text"};
+        std::vector<std::string> missingFields;
+        for (const std::string& field : requiredFields)
+            if (!payload.isMember(field)) missingFields.push_back(field);
+
+        if (!missingFields.empty()) {
+            Json::Value response;
+            std::string error = "Missing required fields";
+            int i = 0;
+            for (const std::string& field : missingFields) {
+                error += field;
+                if (i++ < missingFields.size() - 1) error += ", ";
+            }
+            context.error(error);
+            response["error"] = error;
+            response["ok"] = false;
+            return res.json(response, 400);
+        }
+
+        std::string token = req.headers["authorization"].asString();
+        int space = token.find(" ");
+        token = token.substr(space + 1);
+
+        auto verifier = jwt::verify().allow_algorithm(
+            jwt::algorithm::hs256(std::getenv("VONAGE_API_SIGNATURE_SECRET")));
+
+        auto decoded = jwt::decode(token);
```

Comment:
Are we using `auto` because it's an actual dynamic type coming from `decode` or can we use proper types?

Reply:
I was using `auto` because I was following the example shared by the [https://github.com/Thalhammer/jwt-cpp](https://github.com/Thalhammer/jwt-cpp). 

But, we can change it to types `jwt::verifier<jwt::default_clock, jwt::traits::kazuho_picojson>` for `verifier`, and `jwt::decoded_jwt<jwt::traits::kazuho_picojson>` for decoded

Reply:
can we use specific types please

Reply:
okay, changed it